src/python/pants/option/global_options.py

```diff
@@ -1657,6 +1657,12 @@ class GlobalOptions(BootstrapOptions, Subsystem):
 
             The new semantics will become the default in Pants 2.14, and the old semantics will be
             removed in 2.15.
+
+            This also impacts the behavior of the `tailor` goal. If this option is true,
+            `{bin_name()} tailor` without additional arguments will run over the whole project, and
+            `{bin_name()} tailor dir` will run over `dir` and all recursive sub-directories. If
+            false, you must specify arguments, like `{bin_name()} tailor ::` to run over the
+            whole project; specifying a directory will only add targets for that directory.
```

Comment:
For now, `update-build-files` ignores what the global option is set to and warns if no args are set: https://github.com/pantsbuild/pants/pull/15581. I figured that was simpler to do and valid -- the issue with `tailor` is we have to retcon what a `DirLiteralSpec` means

I don't know if I should add a similar affordance here? The global option still will change the behavior of `dir`. But regardless, if you leave off arguments, we will take that to mean everything, only with a warning. Wdyt?

Reply:
> But regardless, if you leave off arguments, we will take that to mean everything, only with a warning. Wdyt?

Only until the deprecation flag is actually set, which someone will presumably do the instant they see the message. Since it isn't contextual, they likely won't ever see it in the context of `tailor`.